src/keysignature.ts

```diff
@@ -19,115 +19,47 @@ export class KeySignature extends StaveModifier {
   protected keySpec?: string;
   protected alterKeySpec?: string[];
 
-  // Space between natural and following accidental depending
-  // on vertical position
-  static accidentalSpacing: Record<string, { above: number; below: number }> = {
-    '#': {
-      above: 6,
-      below: 4,
-    },
-    b: {
-      above: 4,
-      below: 7,
-    },
-    n: {
-      above: 4,
-      below: 1,
-    },
-    '##': {
-      above: 6,
-      below: 4,
-    },
-    bb: {
-      above: 4,
-      below: 7,
-    },
-    db: {
-      above: 4,
-      below: 7,
-    },
-    d: {
-      above: 4,
-      below: 7,
-    },
-    bbs: {
-      above: 4,
-      below: 7,
-    },
-    '++': {
-      above: 6,
-      below: 4,
-    },
-    '+': {
-      above: 6,
-      below: 4,
-    },
-    '+-': {
-      above: 6,
-      below: 4,
-    },
-    '++-': {
-      above: 6,
-      below: 4,
-    },
-    bs: {
-      above: 4,
-      below: 10,
-    },
-    bss: {
-      above: 4,
-      below: 10,
-    },
-  };
-
   // Create a new Key Signature based on a `keySpec`
   constructor(keySpec: string, cancelKeySpec?: string, alterKeySpec?: string[]) {
     super();
 
     this.setKeySig(keySpec, cancelKeySpec, alterKeySpec);
     this.setPosition(StaveModifierPosition.BEGIN);
-    this.glyphFontScale = Tables.NOTATION_FONT_SCALE;
+    this.glyphFontScale = Tables.lookupMetric('fontSize');
     this.glyphs = [];
-    this.xPositions = []; // relative to this.x
     this.paddingForced = false;
   }
 
   // Add an accidental glyph to the `KeySignature` instance which represents
   // the provided `acc`. If `nextAcc` is also provided, the appropriate
   // spacing will be included in the glyph's position
-  convertToGlyph(acc: { type: string; line: number }, nextAcc: { type: string; line: number }): void {
-    const accGlyphData = Tables.accidentalCodes(acc.type);
-    const glyph = new Glyph(accGlyphData.code, this.glyphFontScale);
+  convertToGlyph(acc: { type: string; line: number }, nextAcc: { type: string; line: number }, stave: Stave): void {
+    const code = Tables.accidentalCodes(acc.type);
+    const glyph = new Element(Category.KeySignature);
+    glyph.setText(code);
+    glyph.measureText();
 
     // Determine spacing between current accidental and the next accidental
-    let extraWidth = 1;
-    if (acc.type === 'n' && nextAcc) {
-      const spacing = KeySignature.accidentalSpacing[nextAcc.type];
-      if (spacing) {
-        const isAbove = nextAcc.line >= acc.line;
-        extraWidth = isAbove ? spacing.above : spacing.below;
-      }
-    }
-
+    const extraWidth = 1;
     // Place the glyph on the stave
-    this.placeGlyphOnLine(glyph, this.checkStave(), acc.line);
+    glyph.setYShift(stave.getYForLine(acc.line));
+    glyph.setXShift(
+      (this.glyphs.length > 0
+        ? this.glyphs[this.glyphs.length - 1].getXShift() + this.glyphs[this.glyphs.length - 1].getWidth() + extraWidth
+        : 0)
+    );
```

Comment:
This part is a little hard to understand.

Can we extract `this.glyphs[this.glyphs.length - 1]` into a local variable called `lastGlyph`? Might make it easier to read.

Reply:
Or should it be `prevGlyph`?? Anyways, choose the best name you can think of for the local variable. I think it will make this part of the code easier to understand.

Reply:
done